PF/Assignment1/Graph.java

```diff
@@ -0,0 +1,132 @@
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+public class Graph {
+
+	// adding connection between two entity
+	public void addEdge(Entity entityFirst, Entity entitySecond) {
+		Iterator<Entity> itr1 = Processing.nodeSet.iterator(); // Iterating list
+																// of questions
+		while (itr1.hasNext()) {
+			if (itr1.next().getEmail().equals(entitySecond.getEmail())) {
+				if (Processing.friendMap.containsKey(entityFirst.getEmail())) {
+					if (!Processing.friendMap.get(entityFirst.getEmail())
+							.contains(entitySecond.getEmail())) {
+						Processing.friendMap.get(entityFirst.getEmail()).add(
+								entitySecond.getEmail());
+						System.out.println(entitySecond.getEmail()
+								+ " is now friend of you");
+					} else if (Processing.friendMap.get(entityFirst.getEmail())
+							.contains(entitySecond.getEmail())) {
+						System.out.println(entitySecond.getEmail()
+								+ " is already friend of you");
+					} else {
+						System.out.println(entitySecond.getEmail()
+								+ " does not exist");
+					}
+				} else {
+					Iterator<Entity> itr2 = Processing.nodeSet.iterator(); // Iterating
```

Comment:
ACTION: There is no need to iterate whole nodes again. you have to just find second entity is already friend or not of first Entity and set a variable isfriend true or false accordingly and outside the loop based on that variable display message or add friend in friend map.